cl/corpus_importer/management/commands/update_casenames_wl_dataset.py

```diff
@@ -0,0 +1,368 @@
+    """
+
+    total_clusters_updated = 0
+    total_dockets_updated = 0
+
+    logger.info(f"Processing {filepath}")
+    df = pd.read_csv(filepath).dropna()
+    for row in df.itertuples():
+        (
+            index,
+            west_case_name,
+            court,
+            date_str,
+            cite1,
+            cite2,
+            docket,
+            volume,
+        ) = row
+
+        clean_docket_num = docket.strip('="').strip('"')
+        if not clean_docket_num:
+            logger.info(f"Row index: {index} - No docket number found.")
+            continue
+
+        date_filed = parse_date(date_str)
+        if not date_filed:
+            logger.info(
+                f"Row index: {index} - No valid date found: {date_str}"
+            )
+            continue
+
+        valid_citations = parse_citations([cite1, cite2])
+
+        if not valid_citations:
+            logger.info(f"Row index: {index} - Missing two valid citations.")
+            continue
+
+        # Query for possible matches using data from row
+        possible_matches = query_possible_matches(
+            valid_citations=valid_citations,
+            docket_number=clean_docket_num,
+            date_filed=date_filed,
+        )
+
+        if not possible_matches:
+            logger.info(f"Row index: {index} - No matches found.")
+            continue
+
+        matches = []
+        for match in possible_matches:
+            cl_case_name = (
+                match.cluster.case_name_full
+                if match.cluster.case_name_full
+                else match.cluster.case_name
+            )
+
+            west_case_name = combine_initials(west_case_name)
+            cl_case_name = combine_initials(cl_case_name)
+
+            case_name_match = check_case_names_match(
+                west_case_name, cl_case_name
+            )
+            if case_name_match:
+                matches.append(match)
+
+        if len(matches) != 1:
+            logger.warning(
+                f"Row index: {index} - Failed, Matches found: {len(matches)} - Matches: {[cluster.id for cluster in matches]}"
+            )
+            continue
+
+        logger.info(
+            f"Row index: {index} - Match found: {matches[0].cluster_id} - Csv case name: {west_case_name}"
+        )
+
+        if dry_run:
+            # Dry run, don't save anything
+            continue
+
+        # Update case names
+        cluster_updated, docket_updated = update_matched_case_name(
+            matches[0].cluster, west_case_name
+        )
+
+        if cluster_updated:
+            total_clusters_updated = +1
+
+        if docket_updated:
+            total_dockets_updated = +1
+
+        # Add any of the citations if possible
+        add_citations_to_cluster(
+            [
+                f"{cite.get('volume')} {cite.get('reporter')} {cite.get('page')}"
+                for cite in valid_citations
+            ],
+            matches[0].cluster_id,
+        )
```

Comment:
id love to reuse this - but I dont think we should - its not as robust and causes use to re-create the citation to parse it again and do a bunch of eyecite.  I think it would be smarter to identify the type when we validate teh citations and just create the citation here.  


something like ... 


```

            matched_cluster = matches[0].cluster

            if dry_run:
                # Dry run, don't save anything
                continue

            # Update case names

            with transaction.atomic():
                cluster_updated, docket_updated = update_matched_case_name(
                matched_cluster, west_case_name
                )

                for citation in valid_citations:
                    if Citation.obects.filter(reporter=citation.get("reporter"), cluster=matched_cluster.id).exists():
                        logger.warning("Can not save mismatched citation")
                        raise("issue with reporter already here... if its not the same one")
                    citation['cluster_id'] = matched_cluster.id
                    Citation.objects.get_or_create(**citation)
```

I think.  

This means the parse citation would look something like this 

```

def parse_citations(citation_strings: list[str]) -> list[dict]:
    """Validate citations with Eyecite.

    :param citation_strings: List of citation strings to validate.
    :return: List of validated citation dictionaries with volume, reporter, and page.
    """
    validated_citations = []

    for cite_str in citation_strings:
        # Get citations from the string
        found_cites = get_citations(cite_str, tokenizer=HYPERSCAN_TOKENIZER)
        if len(found_cites) != 1:
            continue

        citation = found_cites[0]

        if isinstance(citation, FullCaseCitation):
            volume = citation.groups.get("volume")

            # Validate the volume
            if not volume or not volume.isdigit():
                continue

            if not citation[0].corrected_reporter():
                reporter_type = Citation.STATE
            else:
                cite_type_str = citation[0].all_editions[0].reporter.cite_type
                reporter_type = map_reporter_db_cite_type(cite_type_str)

            validated_citations.append(
                {
                    "volume": citation.groups["volume"],
                    "reporter": citation.corrected_reporter(),
                    "page": citation.groups["page"],
                    "type": reporter_type
                }
            )

    return validated_citations
```

I took this from the other function - but im not sure why we need the odd if citation reporter use Citation.State ... ?

Reply:
I have already checked it and I think this part is not necessary:

```
if not citation[0].corrected_reporter():
                reporter_type = Citation.STATE
```

if i remember correctly, we reused that from some old code, i guess we did it that way maybe because reporters-db wasn't complete as it is now.

I also tested if that condition occurs with any of the current citations we have on CL, but got not result, we never got around to that, i think it is safe to remove that part